02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,68 @@
+const { people } = require("../data")
+
+const getPeople = (req, res) => {
+    res.status(200).json(people)
+}
+
+const addPerson = (req, res) => {
+    if (!req.body.name) {
+        return res.status(400).json({ success: false, message: "Please provide a name" });
+    }
+    people.push({ id: people.length + 1, name: req.body.name });
+    res.status(201).json({ success: true, name: req.body.name });
+}
+
+const getPerson = (req, res) => {
+
+    const checkPerson = checkPersonId(req, res)
+
+    if (!checkPerson.success) {
+        return res.status(404).send(checkPerson)
+    }
+
+    res.send(checkPerson.person )
+}
+
+const updatePerson = (req, res) => {
+
+    const checkPerson = checkPersonId(req, res)
+
+    if (!checkPerson.success) {
+        return res.status(404).send(checkPerson)
+    }
+
+    const newPerson = people.map((person) => {
+        if (person.id === Number(req.params.id)) {
+            person.name = req.body.name
+        }
+        return person
+    })
+
+    res.status(200).json({ success: true, data: checkPerson.person, people })
+}
+
+const deletePerson = (req, res) => {
+   
+    const checkPerson = checkPersonId(req, res)
+
+    if (!checkPerson.success) {
+        return res.status(404).send(checkPerson)
+    }
+
+    newPeople = people.filter((person) => person.id !== Number(req.params.id))
+    res.status(200).json({ success: true, message: `Person ${req.body.name} deleted successfully`, newPeople })
```

Comment:
- This operation doesn't work as expected. You're correctly filtering out the person to be deleted but you're not saving it to the right variable, you should update the `people` variable instead of the `newPeople` (which you forgot to declare, too).
    - The `people` variable is a constant, though. The simpler way to go around that and fix this operation that is to change the declaration to use `let`.
    - Another pointer I have for this is that you shouldn't include the complete people list in you response. The request here doesn't impact all resources, just one. So the response should only be related to it.
    - Also, the DELETE request probably won't have the person's name as part of the body. Only the id of the person to be deleted will be passed. You do have the person's name from the `checkPerson` object though.

```javascript
// people declaration
let { people } = require("../data");
```

```suggestion
    const idToDelete = Number(req.params.id);
    people = people.filter((person) => person.id !== idToDelete)
    res.status(200).json({ success: true, message: `Person with id ${idToDelete} and name ${checkPerson.person.name} deleted successfully` })
```